File: Codes/yore/145_post_order_traversal/145_post_order_traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> ans;
        postTree(root, ans);

        return ans;
        
    }

    void postTree(TreeNode* root, vector<int> &ans){
        if(root!=NULL){
            postTree(root->left, ans);
            postTree(root->right, ans);
            ans.push_back(root->val);

        }
    }
};

File: Codes/yore/145_post_order_traversal/145_post_order_traversal.java
public List<Integer> postorderTraversal(TreeNode root) {
    List<Integer> list = new ArrayList<>();
    postOrderRecursion(root, list);
    return list;
}

public void postOrderRecursion(TreeNode node, List<Integer> list) {
    if (node != null) {
        postOrderRecursion(node.left, list);
        postOrderRecursion(node.right, list);
        list.add(node.val);
    }
}